src/truncated/normal.jl

```diff
@@ -128,14 +128,22 @@ function rand(rng::AbstractRNG, d::Truncated{<:Normal{<:Real},Continuous})
     d0 = d.untruncated
     μ = mean(d0)
     σ = std(d0)
+    a, b = extrema(d)
     if isfinite(μ)
-        lower, upper = extrema(d)
-        a = (lower - μ) / σ
-        b = (upper - μ) / σ
-        z = randnt(rng, a, b, d.tp)
+        if iszero(σ)
+            if a <= μ <= b
+                z = 0.0
```

Comment:
I assume this will cause type instabilities, e.g., if parameters of `Normal` and `Truncated` are `Float32`?

Reply:
`z` is always `Float64`, both here and the branch that calls `randnt` (that function requires all inputs be `Float64` and it returns a `Float64`). But if `µ` and the truncation points are wider, say `BigFloat`, then you'll get a `MethodError` from `randnt` when `isfinite(µ) && σ > 0` and a `BigFloat` result otherwise. `Float32` works because `extrema` promotes the truncation points to `Float64`, so every branch will return `Float64`.